packages/roon-kit/src/interfaces.ts

```diff
@@ -259,10 +262,31 @@ export interface RoonApiTransportZones {
 
 export type RoonApiTransportQueueSubscriptionCallback = (response: RoonSubscriptionResponse, body: RoonApiTransportQueue) => void;
 
+export interface QueueItem extends RoonThreeLine{
+    queue_item_id: number;
+    length: number;
+    image_key: string;
+}
+
 export interface RoonApiTransportQueue {
-    items: Item[];
+    items?: QueueItem[];
+    changes?: QueueChange[];
+}
+
+export interface RemoveQueueChange {
+    operation: "remove";
+    index: number;
+    count: number;
+}
+
+export interface InsertQueueChange {
+    operation: "insert";
+    index: number;
+    items: QueueItem[];
 }
 
+export type QueueChange = RemoveQueueChange | InsertQueueChange;
```

Comment:
I haven't done that much `ts` recently, so there might be a more elegant and clever way to type guard `InsertQueueChange` and `RemoveQueueChange` via their `operation` attribute.

Reply:
Found a cleaner solution